File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/location/AnimatorListenerHolder.java
package org.maplibre.android.location;

class AnimatorListenerHolder {
  @MapLibreAnimator.Type
  private final int animatorType;
  private final MapLibreAnimator.AnimationsValueChangeListener listener;

  AnimatorListenerHolder(@MapLibreAnimator.Type int animatorType,
                         MapLibreAnimator.AnimationsValueChangeListener listener) {
    this.animatorType = animatorType;
    this.listener = listener;
  }

  @MapLibreAnimator.Type
  public int getAnimatorType() {
    return animatorType;
  }

  public MapLibreAnimator.AnimationsValueChangeListener getListener() {
    return listener;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    AnimatorListenerHolder that = (AnimatorListenerHolder) o;

    if (animatorType != that.animatorType) {
      return false;
    }
    return listener != null ? listener.equals(that.listener) : that.listener == null;
  }

  @Override
  public int hashCode() {
    int result = animatorType;
    result = 31 * result + (listener != null ? listener.hashCode() : 0);
    return result;
  }
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/AnimatorListenerHolder.kt
package org.maplibre.android.location

import org.maplibre.android.location.MapLibreAnimator.AnimationsValueChangeListener

internal class AnimatorListenerHolder(
    @param:MapLibreAnimator.Type @get:MapLibreAnimator.Type val animatorType: Int,
    val listener: AnimationsValueChangeListener<*>,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }

        other as AnimatorListenerHolder

        if (animatorType != other.animatorType) {
            return false
        }
        return listener == other.listener
    }

    override fun hashCode(): Int {
        var result = animatorType
        result = 31 * result + listener.hashCode()
        return result
    }
}
